service/dap/server.go

```diff
@@ -550,58 +550,66 @@ func (s *Server) onLaunchRequest(request *dap.LaunchRequest) {
 		s.config.Debugger.WorkingDir = wdParsed
 	}
 
-	noDebug := false
-	if noDebugArg, ok := request.Arguments["noDebug"]; ok {
-		if v, ok := noDebugArg.(bool); ok && v { // noDebug == true
-			noDebug = true
-		}
-	}
-
-	var err error
-	if noDebug {
-		if s.noDebugProcess, err = s.launchWithoutDebug(program, targetArgs, s.config.Debugger.WorkingDir); err != nil {
+	if noDebug, _ := request.Arguments["noDebug"].(bool); noDebug {
```

Comment:
Isn't it the recommended practice to check err/ok instead of relying on the default/value not getting populated when an operation fails? Or does that not apply to scalars?
noDebug, ok := request.Arguments["noDebug"].(bool); ok && noDebug {...}

Reply:
I'm not aware of anyone recommending this.

Reply:
Good to know. I remember this coming up in some code review in my early Go days, but it was probably about structs, not scalars.

Reply:
I am using this multi-value returning type assertion here to avoid panic.

According to the language spec, when the type assertion fails, the value is a zero value (false, 0, empty struct). In this case, zero value (false) makes sense, so I don't think this form is violating the spec or any style suggestion. (for example, https://golang.org/src/net/http/server.go#L3147) When accessing a struct field, checking ok may be useful when accessing the zero value's field isn't desirable. Even when accessing a scalar field, checking ok may be necessary when distinction between zero value vs uninitialized value is necessary. All depend on the context.

For errors, yes, it's strongly recommended not to discard error (but as always, there are exceptions).